### DISCUSSION TITLE
Incorrect characters in HTML report of IfcTester

### DISCUSSION THREAD

**Discussion by Zeraxel** [IfcOpenShell/IfcOpenShell#3858] [Q&A] [answered]
I've got problem with displayed characters in HTML  report from ifcTester, started from python script

```
import ifcopenshell
import ifctester
from ifctester import reporter, ids

my_ifc = ifcopenshell.open("D:/ifcOpenShellTest/IfcTester/IDS_TestFile_001.ifc")
my_ids = ids.open("D:/ifcOpenShellTest/IfcTester/IDS_TestIds_001.ids", validate=True)

my_ids.validate(my_ifc)


reporter = ifctester.reporter.Html(my_ids)
reporter.report()
reporter.to_file("D:/ifcOpenShellTest/IfcTester/jest.html")
```

![Screenshot_20231007_140908_Microsoft Remote Desktop](https://github.com/IfcOpenShell/IfcOpenShell/assets/62011988/779797ad-72e2-41dc-9f5a-823b06e5efd2)

My IDS file (and ifc too) have Cyrillic characters. So, i think, this is the main reason. What can i do, to solve this?
I try use blenderBIM , so it doesn't have that problems, Cyrillic characters successfully displayed in HTML report.

**Comment by Moult** [comment]
Are you saying that the Cyrillic characters display fine in the HTML report from Blender but not from the HTML report if you run it via Python script?

**Reply by Zeraxel** [reply]
> Are you saying that the Cyrillic characters display fine in the HTML report from Blender but not from the HTML report if you run it via Python script?

That's right, from blenerBIM i have correctly displayed report.

**Reply by Moult** [reply]
That's bizarre. It's all the same output and the same code. Are you viewing it with the same browser? Is it possible for you to share the file / code with me to try to recreate on my end?

**Reply by Zeraxel** [reply]
> That's bizarre. It's all the same output and the same code. Are you viewing it with the same browser? Is it possible for you to share the file / code with me to try to recreate on my end?

Yes, no problem. https://file.io/apLvuF2oWOsw archive with ifc and ids. It is simple test files.

**Reply by Zeraxel** [reply]
UPD: i found another strange thing. Both of reports from Python Script and from BlenderBIM have the same HTML text (99%, difference only with export time). But from BlenderBIM html report open in browser with Windows-1251 charset, but from Python Srcript it opened as UTF-8 charset. So, As soon as I changing charset to Windows-1251, it was work correctly.
So, What does BlenderBIM do that the report opens correctly, but from py script is not?

**Reply by aothms** [reply, marked as answer]
> What does BlenderBIM do that the report opens correctly, but from py script is not?

Maybe some environment variables related to your language locale and locale. What does `locale` report in both cases? E.g check output of `subprocess.run(['locale'])` from both the blender py console and a standalone python script.

**Reply by Zeraxel** [reply]
> Maybe some environment variables related to your language locale and locale. What does `locale` report in both cases? E.g check output of `subprocess.run(['locale'])` from both the blender py console and a standalone python script.

Thanks. I Use another function to get the locale, and results are different: https://imgur.com/a/xYSZIEU 
Python IDLE: cp1251
blenderBIM: cp65001

**Reply by Zeraxel** [reply]
I found a solve. 

> _Control panel -> Clock and Region -> Region -> Administrative -> Change system locale -> Check Beta: Use Unicode UTF-8 After this, and a reboot, I confirmed that locale.getpreferredencoding () returns ‘cp65001’ (=UTF-8) and that functions like open default to UTF-8._

After that i have correctly report in HTML. Thank you for help @aothms ! Your answer led me to a solution!
